### DISCUSSION TITLE
subcategories should be populated according to selection of category

### DISCUSSION THREAD

**Discussion by gauravsapkal1** [payloadcms/payload#2751] [Q&A]
I am using payloadcms and in collection of posts I have 
    {
      name: "category",
      type: "relationship",
      relationTo: "categories",
      required:true
    }, this category field , now I want to include subcategories field but it should show subcategories only from array which is present in selected category i also have subcategories collection and subcategories present in every category from that collection only

**Comment by DanRibbens** [comment]
Hello @gauravsapkal1,

I think I understand what you are trying to do. Have you looked at `filterOptions` property of relationship fields? You should be able to filter the subcategory down to the only the ones related to the parent by returning a query in your filterOptions function assuming you have a relationship field on subcatory to say which parent it belongs to.

https://payloadcms.com/docs/fields/relationship#filtering-relationship-options

Does that make sense?

Good luck!

**Reply by gauravsapkal1** [reply]
@DanRibbens  My question is I have collection categories and collection subctegories, in categories collection i added subcategories through relationship, so while creating any post i need that my subcategories should populate in option depending upon selection of category. But i am not able to do that so please help me with that

**Comment by Stupidism** [comment]
@gauravsapkal1 I'm using subcategories in this way:

```
{
      type: 'row',
      fields: [
        {
          name: 'state',
          label: 'State',
          type: 'relationship',
          relationTo: 'states',
          validate: isNotEmpty,
          admin: {
            width: '50%',
            condition: resourceFieldCondition('state'),
          },
        },
        {
          name: 'city',
          label: 'City',
          type: 'relationship',
          relationTo: 'cities',
          validate: isNotEmpty,
          filterOptions: ({ data }) => {
            return {
              state: { equals: data?.state },
            };
          },
          admin: {
            width: '50%',
            condition: resourceFieldCondition('city'),
          },
        },
      ],
    },
